internal/config/ociartifact/ociartifact.go

```diff
@@ -121,15 +152,115 @@ func (o *OCIArtifact) Pull(ctx context.Context, img string, opts *PullOptions) (
 		return nil, fmt.Errorf("read with limit: %w", err)
 	}
 
-	if err := verifyDigest(&layer, layerBytes); err != nil {
+	if err := verifyDigest(layer, layerBytes); err != nil {
 		return nil, fmt.Errorf("verify digest of layer: %w", err)
 	}
 
+	keyPath := o.tryWriteCache(ctx, useCache, opts, layer, layerBytes)
+
 	return &Artifact{
 		Data: layerBytes,
+		Path: keyPath,
 	}, nil
 }
 
+func (o *OCIArtifact) prepareCache(ctx context.Context, opts *PullOptions) (useCache bool) {
+	if opts.CachePath == "" {
+		return false
+	}
+
+	if err := o.impl.MkdirAll(opts.CachePath, 0o700); err != nil {
+		log.Errorf(ctx, "Unable to create cache path: %v", err)
+		return false
+	}
+
+	entries, err := o.impl.ReadDir(opts.CachePath)
+	if err != nil {
+		log.Errorf(ctx, "Unable to read cache path: %v", err)
+		return false
+	}
+
+	maxAge := opts.CacheEntryMaxAge
+	if opts.CacheEntryMaxAge == 0 {
+		maxAge = defaultCacheEntryMaxAge
+	}
+
+	now := time.Now()
+	if maxAge > 0 {
+		for _, entry := range entries {
+			info, err := entry.Info()
+			if err != nil {
+				log.Errorf(ctx, "Unable to get cache item info for %q: %v", entry.Name(), err)
+				continue
+			}
+
+			if diff := now.Sub(info.ModTime()); diff > maxAge {
+				p := filepath.Join(opts.CachePath, info.Name())
+				log.Infof(ctx, "Deleting old cache item %q (age: %v)", p, diff.Round(time.Second))
```

Comment:
Nit. If we want to be consistent with:

```golang
log.Infof(ctx, "Removing cached artifact layer for digest %q", layer.BlobInfo.Digest)
```

However, perhaps using "Deleting" in both places would be better sounding?

```suggestion
				log.Infof(ctx, "Removing old cache item %q (age: %v)", p, diff.Round(time.Second))
```

Reply:
Also, should we call things "cache item" or "artifact layer" throughout? What would be more appropriate?

Thoughts?

Reply:
Fixed, went for `artifact layer` all over the place.